### PR TITLE
bugfix: Print correct method signature for Selectable

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would not show anything when selectDynamic or applyDynamic methods were used. Now, we look up the refinement to get the proper signatures.

Fixes https://github.com/scalameta/metals/issues/4192

### CODE REVIEW

**Review by prolativ — APPROVED** [review]

**Review by tanishiking — APPROVED** [review]
LGTM 🥳

File: mtags/src/main/scala-3/scala/meta/internal/pc/HoverProvider.scala [hunk 1]
```diff
@@ -129,6 +132,35 @@ object HoverProvider:
...
     private def isPoint: Boolean = pos.start == pos.end
 
+  private def fallbackToDynamics(
+      path: List[Tree],
+      printer: MetalsPrinter,
+  )(using Context): ju.Optional[Hover] = path match
+    case (head @ Apply(
+          Select(sel, n),
+          List(Literal(Constant(name: String))),
+        )) :: _ if n == nme.selectDynamic || n == nme.applyDynamic =>
```
Comment (dos65) [inline mtags/src/main/scala-3/scala/meta/internal/pc/HoverProvider.scala:142]:
```suggestion
    case Apply(
          Select(sel, n),
          List(Literal(Constant(name: String))),
        ) :: _ if n == nme.selectDynamic || n == nme.applyDynamic =>
```

**Review by dos65 — APPROVED** [review]
LGTM!

File: mtags/src/main/scala-3/scala/meta/internal/pc/HoverProvider.scala (hunk 1 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/HoverProvider.scala]: Fixed! Good catch!

**Pull request merged**